carbon-design-system/carbon
# refactor(react): update InlineCheckbox to FC

Reference #10281 

This PR updates our `InlineCheckbox` internal component to a Functional Component.

#### Changelog

**New**

**Changed**

- Update `InlineCheckbox` to FC

**Removed**

#### Testing / Reviewing

- Verify all places where `InlineCheckbox` is used work as expected in storybook and in tests
  - TableSelectRow
  - TableSelectAll
- Verify tests pass as expected
- Verify component refactor matches our style guidelines
